plugins/modules/aci_interface_policy_spine_policy_group.py

```diff
@@ -0,0 +1,351 @@
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        policy_group=dict(type="str", aliases=["policy_group_name", "spine_policy_group_name", "name"]),  # Not required for querying all objects
+        description=dict(type="str", aliases=["descr"]),
+        name_alias=dict(type="str"),
+        link_level_policy=dict(type="str", aliases=["link_level_policy_name"]),
+        link_flap_policy=dict(type="str", aliases=["link_flap_policy_name"]),
+        cdp_policy=dict(type="str", aliases=["cdp_policy_name"]),
+        mac_sec_policy=dict(type="str", aliases=["mac_sec_policy_name"]),
+        attached_entity_profile=dict(type="str", aliases=["aep_name", "aep"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["policy_group"]],
+            ["state", "present", ["policy_group"]],
+        ],
+    )
+
+    policy_group = module.params.get("policy_group")
+    description = module.params.get("description")
+    name_alias = module.params.get("name_alias")
+    link_level_policy = module.params.get("link_level_policy")
+    link_flap_policy = module.params.get("link_flap_policy")
+    cdp_policy = module.params.get("cdp_policy")
+    mac_sec_policy = module.params.get("mac_sec_policy")
+    attached_entity_profile = module.params.get("attached_entity_profile")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    class_config = dict(
+        name=policy_group,
+        descr=description,
+        nameAlias=name_alias,
+    )
+
+    child_configs = [
+        dict(
+            infraRsHIfPol=dict(
+                attributes=dict(
+                    tnFabricHIfPolName=link_level_policy,
+                ),
+            ),
+        ),
+        dict(
+            infraRsLinkFlapPol=dict(
+                attributes=dict(
+                    tnFabricLinkFlapPolName=link_flap_policy,
+                ),
+            ),
+        ),
+        dict(
+            infraRsCdpIfPol=dict(
+                attributes=dict(
+                    tnCdpIfPolName=cdp_policy,
+                ),
+            ),
+        ),
+        dict(
+            infraRsMacsecIfPol=dict(
+                attributes=dict(
+                    tnMacsecIfPolName=mac_sec_policy,
+                ),
+            ),
+        ),
+        dict(
+            infraRsAttEntP=dict(
+                attributes=dict(
+                    tDn="uni/infra/attentp-{0}".format(attached_entity_profile),
+                ),
+            ),
+        ),
```

Comment:
Put this one behind a condition if attached_entity_profile is not None

Reply:
resolved